### PR TITLE
feat(plugin): Adds alias-printer plugin

### PR DESCRIPTION
**Pull request by ndrone**
This plugin searches the defined aliases and outputs any command that matches the inputted alias. So that when presenting the audience understands the alias being executed.

## Standards checklist:

- [ ] The PR title is descriptive.
- [ ] The PR doesn't replicate another PR which is already open.
- [ ] I have read the contribution guide and followed all the instructions.
- [ ] The code follows the code style guide detailed in the wiki.
- [ ] The code is mine or it's from somewhere with an MIT-compatible license.
- [ ] The code is efficient, to the best of my ability, and does not waste computer resources.
- [ ] The code is stable and I have tested it myself, to the best of my abilities.

## Changes:

- [...]

## Other comments:

...

### CODE REVIEW

**Comment by carlosala** [comment]
There are some things to improve in this plugin, but I don't think this is really important, as we have other ways to get same information. When you enable the hook it always prints the information whether or not that is an alias, and this is a problem in my opinion.
I think it's better to run `alias <your_alias>` to get same information, and only when you really need it. For example:
```zsh
$ alias gl
gl='git pull'
```
What do you think about?

**Comment by ndrone** [comment]
I like it. I'll get the PR updated with the change.

**Comment by carlosala** [comment]
The idea is that `alias` is included in `zsh`, so I feel there's no need of having a plugin doing that. Commonly, people just want to use aliases, and if there's someone that they want to get more information of, just type `alias <thatone>`. I feel a bit verbose and redundant to get the information every time.

**Comment by ndrone** [comment]
Generally I agree with you. This plugin covers a specific scenario. That when I'm presenting, or teaching. That the audience then can understand what the aliases are then executing.

File: plugins/alias-printer/alias-printer.plugin.zsh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      * )
+        # eval doesn't like command chaining '&&' so removing '&&' so that we
+        # generate an array and then loop through the commands and print the
+        # aliases
+        words=$(sed 's/&&//g' <<< $1)
+        eval "words=($words)"
+        for word in $words; do
+          alias "$word"
+        done
+        break
```
[the remark this answers is not in the document]
Comment (mcornella) [reply plugins/alias-printer/alias-printer.plugin.zsh:23]: Also eval-ing words without quoting is command-injection level of bad (think what would happen with a parenthesis or a semicolon in the command).

**Comment by mcornella** [comment]
You don't check whether the first word is an alias. Also, since you're using a for to iterate over the arguments passed, this will display `alias-printer: <command-expanded>` 3 times when called as a preexec hook.

**Comment by ndrone-kr** [comment]
> You don't check whether the first word is an alias. Also, since you're using a for to iterate over the arguments passed, this will display `alias-printer: <command-expanded>` 3 times when called as a preexec hook.

the `break` exits the loop after the first execution

**Comment by mcornella** [comment]
> the `break` exits the loop after the first execution

That's true. To check whether the first word of the command is an alias, you can use something like https://github.com/ohmyzsh/ohmyzsh/blob/957dca698cd0a0cafc6d2551eeff19fe223f41bd/plugins/sudo/sudo.plugin.zsh#L61
to get the first word, and test if it exists in `$aliases` with `(( ${+aliases[$cmd]} ))`.

Make sure to declare the variables you use with `local`.

**Comment by ndrone-kr** [comment]
ok, I'll get it updated later today. Thanks!

**Pull request closed without merging**